src/redux/settings.ts

```diff
@@ -25,7 +26,51 @@ const SETTINGS_UPDATE_NETWORK_SUCCESS =
   'settings/SETTINGS_UPDATE_NETWORK_SUCCESS';
 
 // -- Actions --------------------------------------------------------------- //
-export const settingsLoadState = () => async dispatch => {
+
+/**
+ * The current `settings` state.
+ */
+interface SettingsState {
+  accountAddress: string;
+  chainId: number;
+  language: string;
+  nativeCurrency: string;
+  network: Network;
+}
+
+/**
+ * A `settings` Redux action.
+ */
+type SettingsStateUpdateAction =
+  | SettingsStateUpdateSettingsAddressAction
+  | SettingsStateUpdateNativeCurrencySuccessAction
+  | SettingsStateUpdateNetworkSuccessAction
+  | SettingsStateUpdateLanguageSuccessAction;
+
+interface SettingsStateUpdateSettingsAddressAction {
+  type: typeof SETTINGS_UPDATE_SETTINGS_ADDRESS;
+  payload: SettingsState['accountAddress'];
+}
+
+interface SettingsStateUpdateNativeCurrencySuccessAction {
+  type: typeof SETTINGS_UPDATE_NATIVE_CURRENCY_SUCCESS;
+  payload: SettingsState['nativeCurrency'];
+}
+
+interface SettingsStateUpdateNetworkSuccessAction {
+  type: typeof SETTINGS_UPDATE_NETWORK_SUCCESS;
+  payload: {
+    chainId: SettingsState['chainId'];
+    network: SettingsState['network'];
+  };
+}
+
+interface SettingsStateUpdateLanguageSuccessAction {
+  type: typeof SETTINGS_UPDATE_LANGUAGE_SUCCESS;
+  payload: SettingsState['language'];
+}
+
+export const settingsLoadState = () => async (dispatch: AppDispatch) => {
```

Comment:
Same issue I noticed here: https://github.com/rainbow-me/rainbow/pull/2505#discussion_r743664238

tl;dr `AppDispatch` is being resolved as `AppDispatch<any>`, so it might be worth explicitly typing the actions here with `Dispatch<SettingsStateUpdateAction>`.

The tricky part is that some actions being dispatched in this file are coming from JS files so you might want to annotate those dispatches with `// @ts-expect-error` comments — that way you'll be prompted to come back and clean up this file when those files are converted to TS.